common/setups/rasr/util/decode.py

```diff
@@ -0,0 +1,132 @@
+__all__ = [
+    "RecognitionParameters",
+    "PriorPath",
+    "SearchJobArgs",
+    "Lattice2CtmArgs",
+    "StmArgs",
+    "ScliteScorerArgs",
+    "OptimizeJobArgs",
+]
+
+from dataclasses import dataclass
+from itertools import product
+from typing import Any, Dict, List, Optional, Tuple, TypedDict, Union
+
+from sisyphus import tk
+
+import i6_core.rasr as rasr
+
+from ..config.am_config import Tdp
+
+
+class RecognitionParameters:
+    def __init__(
+        self,
+        *,
+        am_scales: List[float],
+        lm_scales: List[float],
+        prior_scales: List[float],
+        pronunciation_scales: List[Optional[float]],
+        tdp_scales: List[float],
+        speech_tdps: List[Tdp],
+        silence_tdps: List[Tdp],
+        nonspeech_tdps: List[Optional[Tdp]],
+        altas: List[Optional[float]],
+    ):
+        self.am_scales = am_scales
+        self.lm_scales = lm_scales
+        self.prior_scales = prior_scales
+        self.pronunciation_scales = pronunciation_scales
+        self.tdp_scales = tdp_scales
+        self.speech_tdps = speech_tdps
+        self.silence_tdps = silence_tdps
+        self.nonspeech_tdps = nonspeech_tdps
+        self.altas = altas
+
+    def _get_iter(self):
+        return [
+            (am, lm, pri, pron, tdp, speech, sil, nonspeech, al)
+            for am, lm, pri, pron, tdp, speech, sil, nonspeech, al in product(
+                self.am_scales,
+                self.lm_scales,
+                self.prior_scales,
+                self.pronunciation_scales,
+                self.tdp_scales,
+                self.speech_tdps,
+                self.silence_tdps,
+                self.nonspeech_tdps,
+                self.altas,
+            )
+        ]
+
+    def __iter__(self):
+        self.iterations = self._get_iter()
+        return self
+
+    def __next__(self):
+        if self.iterations:
+            return self.iterations.pop(0)
+        else:
+            raise StopIteration
+
+
+@dataclass()
+class PriorPath:
+    prior_xml_path: tk.Path
+    acoustic_mixture_path: tk.Path
+
+
+class LookaheadOptions(TypedDict):
+    test: str  # TODO
+
+
+class SearchJobArgs(TypedDict):
+    use_gpu: bool
+    rtf: float
+    mem: int
+    cpu: int
+    extra_config: Optional[rasr.RasrConfig]
+    extra_post_config: Optional[rasr.RasrConfig]
+
+
+class AdvTreeSearchJobArgs(SearchJobArgs):
```

Comment:
what distinguishes between `AdvTreeSearchJobArgs` and `SearchJobArgs`.. do we want to have e.g. `LabelSyncSearchJobArgs` in the future?

but then I guess _all_ Search Jobs will have `search_parameters` (so it should go into the base class)?..